tests/acceptance/stepDefinitions/publicLinkContext.js

```diff
@@ -5,47 +5,149 @@ const sharingHelper = require('../helpers/sharingHelper')
+  })
+
+Then('the user should see an error message on the public link share dialog saying {string}',
+  function (errorMessage) {
+    return client.page.FilesPageElement
+      .publicLinksDialog()
+      .assertErrorMessageForPublicLinks(errorMessage)
+  })
+
+Then('as user {string} the folder {string} should not have any public link', async function (sharer, resource) {
+  const publicLinkShares = await sharingHelper.getAllPublicLinkShares(sharer)
+  let found = false
+  for (const share of publicLinkShares) {
+    if (share.path === '/' + resource && share.share_type === 3 && share.uid_owner === sharer) {
+      found = true
+      break
+    }
+  }
+  assert.strictEqual(
+    found, false, 'Expected share with user ' + sharer +
+    ' and resource ' + resource + ' is present!\n' + JSON.stringify(publicLinkShares)
+  )
+})
+
+Then('the public should not get access to the publicly shared file', function () {
+  return client
+    .page
+    .publicLinkPasswordPage()
+    .assertResourceAccessDenied()
+})
+
+When('the user tries to edit the public link named {string} of file/folder/resource {string} changing following',
+  async function (linkName, resource, dataTable) {
+    const editData = dataTable.rowsHash()
+    await client.page.FilesPageElement
+      .filesList()
+      .closeSidebar(100)
+      .openPublicLinkDialog(resource)
+    return client.page.FilesPageElement.publicLinksDialog()
+      .editPublicLink(linkName, editData, false)
+  })
+
+When('the user edits the public link named {string} of file/folder/resource {string} changing following',
+  async function (linkName, resource, dataTable) {
+    const editData = dataTable.rowsHash()
+    await client.page.FilesPageElement
+      .filesList()
+      .closeSidebar(100)
+      .openPublicLinkDialog(resource)
+    return client.page.FilesPageElement.publicLinksDialog()
+      .editPublicLink(linkName, editData, true)
+  })
+
+When('the user tries to edit expiration of the public link named {string} of file {string} to past date {string}',
+  async function (linkName, resource, pastDate) {
+    await client.page.FilesPageElement
+      .filesList()
+      .closeSidebar(100)
+      .openPublicLinkDialog(resource)
+    return client.page.FilesPageElement.publicLinksDialog()
+      .assertDisabledExpiryDate(linkName, pastDate)
+  })
+
+When('the user {string} removes the public link named {string} of file/folder/resource {string} using the webUI',
+  async function (sharer, linkName, resource) {
+    await client.page
+      .FilesPageElement
+      .filesList()
+      .closeSidebar(100)
+      .openPublicLinkDialog(resource)
+    return client.page.FilesPageElement.publicLinksDialog()
+      .removePublicLink(linkName)
+  })
+
+Then('public link named {string} should not be listed on the public links list on the webUI', async function (linkName) {
+  await client.page.FilesPageElement.publicLinksDialog()
```

Comment:
Also, remember to remove `async`
```suggestion
  return client.page.FilesPageElement.publicLinksDialog()
```